gtk/docs.md

```diff
@@ -1 +1,353 @@
+to the entire string (i.e. unless you set the range of each attribute
+to [0, `G_MAXINT`)). The reason is that specifying the start_index and
+end_index for a `pango::Attribute` requires knowledge of the exact string
+being displayed, so translations will cause problems.
+
+# Selectable labels
+
+Labels can be made selectable with `Label::set_selectable`.
+Selectable labels allow the user to copy the label contents to
+the clipboard. Only labels that contain useful-to-copy information
+— such as error messages — should be made selectable.
+
+# Text layout # {`label`-text-layout}
+
+A label can contain any number of paragraphs, but will have
+performance problems if it contains more than a small number.
+Paragraphs are separated by newlines or other paragraph separators
+understood by Pango.
+
+Labels can automatically wrap text if you call
+`Label::set_line_wrap`.
+
+`Label::set_justify` sets how the lines in a label align
+with one another. If you want to set how the label as a whole
+aligns in its available space, see the `Widget::halign` and
+`Widget:valign` properties.
+
+The `Label:width-chars` and `Label:max-width-chars` properties
+can be used to control the size allocation of ellipsized or wrapped
+labels. For ellipsizing labels, if either is specified (and less
+than the actual text size), it is used as the minimum width, and the actual
+text size is used as the natural width of the label. For wrapping labels,
+width-chars is used as the minimum width, if specified, and max-width-chars
+is used as the natural width. Even if max-width-chars specified, wrapping
+labels will be rewrapped to use all of the available width.
+
+Note that the interpretation of `Label:width-chars` and
+`Label:max-width-chars` has changed a bit with the introduction of
+[width-for-height geometry management.](geometry-management)
+
+# Links
+
+Since 2.18, GTK+ supports markup for clickable hyperlinks in addition
+to regular Pango markup. The markup for links is borrowed from HTML,
+using the `<a>` with “href“ and “title“ attributes. GTK+ renders links
+similar to the way they appear in web browsers, with colored, underlined
+text. The “title“ attribute is displayed as a tooltip on the link.
+
+An example looks like this:
+
+```no_run
+    use gtk::Label;
+
+    let label = Label::new(None);
+    let text =
+r#"Go to the
+ <a href="http://www.gtk.org title="&lt;i&gt;Our&lt;/i&gt; website">
+ GTK+ website</a> for more..."#;
+    label_set_markup(text);
+```
+
+It is possible to implement custom handling for links and their tooltips with
+the `Label::activate-link` signal and the `Label::get_current_uri` function.
+
+# Implements
+
+[`MiscExt`](trait.MiscExt.html), [`WidgetExt`](trait.WidgetExt.html)
+<!-- impl Label::fn get_use_markup -->
+Returns whether the label’s text is interpreted as marked up with
+the [Pango text markup language](https://developer.gnome.org/pygtk/stable/pango-markup-language.html).
+See gtk_label_set_use_markup ().
```

Comment:
This looks like a botched reference to the `set_use_markup` method in the upstream docs.